src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/ISignatureProvider.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Stratis.FederatedPeg.Features.FederationGateway.Interfaces
+{
+    public interface ISignatureProvider
```

Comment:
Please describe the purpose and behaviour of this class in details, its clear it provides signatures but from who and how are the keys provided?

Reply:
The keys are provided by the API call that activates federation. The purpose of the class is in the associated issue. I will add the detail to the class.